### PR TITLE
fix: attempt to fix LCP in case of audience 

### PR DESCRIPTION
**Pull request by kptdobe**
Managed to improve the LH from 76 to 99 for the flyer page.

Main changes:
- audiences (desktop vs mobile sections) - remove the non matching audiences from the DOM. This helps not processing the desktop audience while you are on mobile.
- LCP candidate - template list is an LCP block - major issue is that the LCP candidate is not the first image (the Express logo) but the first image of the template list. Hardcoded it for now but would require some evolution in the block concept to generalise it (for example, the block decorate could return the LCP candidate if it is not the default one...).

This changes seems to improve a lot of pages (almost back to 100 on several of them)

Test URLs:

Before:
- https://main--express-website--adobe.hlx.live/express/create/flyer?lighthouse=on

After:
- https://fix-audience-lcp--express-website--adobe.hlx.live/express/create/flyer?lighthouse=on

Controlling other pages:
- https://fix-audience-lcp--express-website--adobe.hlx.live/express/?lighthouse=on
- https://fix-audience-lcp--express-website--adobe.hlx.live/express/templates/flyer/christmas?lighthouse=on
- https://fix-audience-lcp--express-website--adobe.hlx.live/express/feature?lighthouse=on
- https://fix-audience-lcp--express-website--adobe.hlx.live/express/feature/image/remove-background?lighthouse=on
- https://fix-audience-lcp--express-website--adobe.hlx.live/express/nonprofits?lighthouse=on
- https://fix-audience-lcp--express-website--adobe.hlx.live/education/express/deployment?lighthouse=on
- https://fix-audience-lcp--express-website--adobe.hlx.live/drafts/ernest_sallee/DOTCOM-73636/content-scheduler?lighthouse=on
- https://fix-audience-lcp--express-website--adobe.hlx.live/drafts/qiyundai/collaboration-hub?lighthouse=on
- https://fix-audience-lcp--express-website--adobe.hlx.live/express/create/flyer?lighthouse=on
- https://fix-audience-lcp--express-website--adobe.hlx.live/express/experiments/ccx0074/test?lighthouse=on
- https://fix-audience-lcp--express-website--adobe.hlx.live/express/experiments/ccx0074/test?lighthouse=on&experiment=ccx0074%2Fchallenger-2

### CODE REVIEW

**Comment by kptdobe** [comment]
I do not think we'll get this one right one day.... the Template List block as LCP block in the context of an audience would require a lot of changes...

**Comment by kptdobe** [comment]
Humm... pagespeed gives me 92 while helix-bot says 83 for the same page and the same code...

image

File: express/scripts/scripts.js [hunk 1]
```diff
@@ -331,6 +331,21 @@ export function readBlockConfig($block) {
...
+ * @return {boolean} is member of this audience
+ */
+function checkExperimentAudience(audience) {
+  if (audience === 'mobile') {
+    return window.innerWidth < 600;
+  }
+  if (audience === 'desktop') {
+    return window.innerWidth > 600;
+  }
+  return true;
```
Comment (rofe) [inline express/scripts/scripts.js:346]: This changes the current implementation from user agent based to window width based, which will impact existing content intended for mobile `devices` (e.g. tablets) which should see the same content as mobiles, but have overlapping window widths with desktops...

Comment (kptdobe) [reply express/scripts/scripts.js]: I did not introduce this method (just moved it from the bottom of the file to the top) but I still broke the logic by re-using this method in the wrong place. Fixing now.

**Review by rofe — APPROVED** [review]

**Review by davidnuescheler — APPROVED** [review]

**Pull request merged**